app/models/alchemy/element/presenters.rb

```diff
@@ -43,7 +43,11 @@ def display_name
     #   Length of characters after the text will be cut off.
     #
     def preview_text(maxlength = 30)
-      (contents.detect(&:preview_content?) || contents.first).try(:preview_text, maxlength)
+      if nestable_elements.any? && !contents.detect(&:preview_content?)
```

Comment:
I think this could be improved, if we split this in to two private methods:

1. `preview_text_from_nested_elements`
2. `preview_text_from_preview_content`

Then we can refactor this method into:

```ruby
preview_text_from_nested_elements ||
  preview_text_from_preview_content
```

Also, we should refactor `contents.detect(&:preview_content?)` into its own method that caches the result, so it's no performance degradation if we call this twice. What we actually are doing.